packages/loader/container-loader/src/prefetchDocumentStorageService.ts

```diff
@@ -36,18 +44,18 @@ export class PrefetchDocumentStorageService extends DocumentStorageServiceProxy
         this.prefetchCache.clear();
     }
 
-    // eslint-disable-next-line @typescript-eslint/promise-function-async
-    private cachedRead(blobId: string): Promise<string> {
+    private async cachedRead(blobId: string): Promise<ArrayBufferLike> {
         if (this.prefetchEnabled) {
-            const prefetchedBlobP: Promise<string> | undefined = this.prefetchCache.get(blobId);
+            const prefetchedBlobP: ArrayBufferLike | undefined = this.prefetchCache.get(blobId);
             if (prefetchedBlobP !== undefined) {
                 return prefetchedBlobP;
             }
-            const prefetchedBlobPFromStorage = this.internalStorageService.read(blobId);
+            const blob = await this.internalStorageService.readBlob(blobId);
```

Comment:
await here changes a bit how this class works, and it may be material.
Before changes, call to this function was synchronous, i.e. it would return right away filling in promise into this.prefetchCache. If another call would come in for same blob right away, it would return same (potentially incomplete) promise, but it would not make another call to storage.

After your changes, we block on await for some amount of time. If during this wait another call for same blob comes in, it will make another call to storage for same blob,

Reverting for this.prefetchCache to contain promises, not actual values is required here to keep existing behavior, which is more correct.